### PR TITLE
UWP visualizer app improvements

### PR DESCRIPTION
**Pull request by anbare**
Massive improvement to the UWP Visualizer app

- Tabbed viewing of files! Tabs are on the left though cause it’s cool and different
- Warnings and errors displayed!
- Remembers which files you had open after relaunch
- Instant preview of card, rather than having to click generate
- Can save changes on a file right from the app

In a future version we need to...

- Ability to edit host config
- Ability to save a newly created file
- Ability to save over the provided sample files (saving into an AppData folder instead)
- Remember which specific tab was last open and restore that on relaunch
- Change the color of the window title bar so it matches the top command bar

But I'm going on vacation so want to get these major improvements in now! Screenshots...

image

image

### CODE REVIEW

**Comment by matthidinger** [comment]
Literally :fire:

File: source/uwp/Visualizer/XamlCardVisualizer.csproj [hunk 1]
```diff
@@ -39,22 +40,66 @@
...
     </Compile>
+    <Compile Include="Converters\ErrorViewModelConverters.cs" />
+    <Compile Include="Converters\NotNullToVisibilityConverter.cs" />
+    <Compile Include="DocumentView.xaml.cs">
+      <DependentUpon>DocumentView.xaml</DependentUpon>
+    </Compile>
+    <Compile Include="Helpers\BindableBase.cs" />
+    <Compile Include="Helpers\IListExtensions.cs" />
+    <Compile Include="Helpers\PayloadValidator.cs" />
+    <Compile Include="Helpers\WeakEventHandler.cs" />
```
Comment (riarenas) [inline source/uwp/Visualizer/XamlCardVisualizer.csproj]: ><Compile Include="Helpers\WeakEventHandler.cs" /> [](start = 4, length = 49)

This file is missing

**Comment by riarenas** [comment]
Build failed with:
Error : Your project is not referencing the "UAP,Version=v10.0.10586" framework. Add a reference to "UAP,Version=v10.0.10586" in the "frameworks" section of your project.json, and then re-run NuGet restore.

**Comment by anbare** [comment]
Looks like it might be an issue with the build server not using NuGet 4.1.0? https://github.com/NuGet/Home/issues/4515

**Comment by riarenas** [comment]
@nesalang Please look into this.

**Comment by Assaya** [comment]
lol

**Comment by Assaya** [comment]
Fiverr
    
    
            
      {
          "@context": "http://schema.org",
          "@type": "EmailMessage",
          "action": {
            "@type": "ViewAction",
            "url": "https://www.fiverr.com/s2/357a8ee75d",
            "name": "Browse Services on Fiverr"
          },
          "description": "Browse Services on Fiverr"
      }
    

        
            
                
                    
                        
                            
                                
                                    
                                        
                                    
                                
                                
                                    
                                    
                                
                                
                                    

                                        
    

    
        Angela is tired of keeping Fiverr a secret.
    

    
        Discover the easiest way to hire online freelancers and get 20% off your first order. From marketing to tech, complete all your projects on Fiverr. On budget. On time.
    

    
        Sounds too good to be true? See for yourself!
    



                                        
                                            
                                                
                                                    
                                                        
                                                            Browse Services On Fiverr
                                                        
                                                    
                                                
                                                
                                                    
                                                        

		
			
			
			The Fiverr Team
		
                                                    
                                                
                                            
                                        
                                    
                                
                            
                        
                    
                
                
                    
                        
                            
                                

                                    © Fiverr International Limited 2010-2019, Fiverr Inc. | 401 Broadway, Suite 1600 New York NY, 10013 USA.
                                    Your personal information will not be stored or used for any marketing purposes.
                                
                            
                         
                    
                
            
        
            

    
If you would like to unsubscribe and stop receiving these emails click here.

**Comment by Assaya** [comment]
When doing the school register I was always last on the list being 'W'. Imagine what Mark Z. thought being the very last at Harvard.
Luckily I had thoughtful teachers and sometimes they would turn the register around and instead of being last on the list, I was first.
I made Mark Z. first on my list.

Lesson learnt - Don't judge a book by it's cover but by the ending.


angela@thehelm

[cid:74fc3c12-75e9-4510-bb07-906fa0d61a3c]

**Pull request merged**